### ISSUE TITLE
Add a classical structural causal model counterfactual example

### ISSUE THREAD

**Issue by eb8680** [BasisResearch/chirho#204]
Such an example should exercise `chirho.constrained.handlers.soft_conditioning`.

**Comment by SamWitty** [comment]
Is this covered by the deep SCM tutorial?

**Comment by eb8680** [comment]
No, I think the point of this would be to demonstrate how to use ChiRho and especially the machinery in `chirho.observational.handlers.soft_conditioning` to directly represent the toy model examples of counterfactuals that Pearl uses so heavily in his writings, like the linear-Gaussian econometrics example in chapter 7.2.1 of Causality. The deep SCM tutorial has too many other complicating factors to fulfill that role.
